JoinColony/colonyCDapp
# Funding an expenditure

## Description

The main focus of this PR is adding the ability to fund an expenditure. It also includes other dependencies to make that possible, and because of that it's grown a bit big in size - apologies. Many commits from the PR have already been reviewed as part of #913 and #923.

## Testing
At this point, it'd be good if testing resembles nearly end-to-end user paths. You should be able to:
- Create an expenditure with an arbitrary number of recipients, with payouts in different tokens
- See the current expenditure pot balance for every token it has payouts in
- Lock the expenditure and see the status update
- While locked, fund the expenditure - this should move funds from the Root domain for all the tokens there are payouts in, and update the balances
- Following funding, finalize the expenditure (the expenditure status should change to "Finalized" after refreshing the page)

Resolves #897
